### PR TITLE
deleting a file when 2 files exist with different case

### PR DESCRIPTION
**Pull request by davitol**
## Related Issue
https://github.com/owncloud/core/issues/32357 part

## Motivation and Context
Spread Tests

## How Has This Been Tested?
Local test runs

### CODE REVIEW

File: tests/acceptance/features/apiWebdavOperations/deleteFile.feature [hunk 1]
```diff
@@ -18,3 +18,16 @@ Feature: delete file
       | dav_version |
       | old         |
       | new         |
+
+  Scenario Outline: delete a file when 2 files exist with different case
+    Given using <dav_version> DAV path
+    When user "user0" uploads file with content "uploaded content" to "/textfile1.txt" using the WebDAV API
```
Comment (individual-it) [inline tests/acceptance/features/apiWebdavOperations/deleteFile.feature]: I suggest to use Given here
Given user "user0" has uploaded ....
as this is a setup step for the tests

File: tests/acceptance/features/apiWebdavOperations/deleteFile.feature [hunk 2]
```diff
@@ -18,3 +18,16 @@ Feature: delete file
...
+    And  user "user0" uploads file with content "uploaded content" to "/TextFile1.txt" using the WebDAV API
+    When user "user0" deletes file "/textfile0.txt" using the WebDAV API
```
Comment (individual-it) [inline tests/acceptance/features/apiWebdavOperations/deleteFile.feature]: do you meant to delete `textfile1.txt`? as we want to test the different cases

File: tests/acceptance/features/apiWebdavOperations/deleteFile.feature [hunk 3]
```diff
@@ -18,3 +18,16 @@ Feature: delete file
...
+    Then the HTTP status code should be "204"
+    And as "user0" file "/textfile0.txt" should not exist
```
Comment (individual-it) [inline tests/acceptance/features/apiWebdavOperations/deleteFile.feature]: see comment above

**Review by individual-it — CHANGES_REQUESTED** [review]

File: tests/acceptance/features/apiWebdavOperations/deleteFile.feature [hunk 4]
```diff
@@ -18,3 +18,15 @@ Feature: delete file
       | dav_version |
       | old         |
       | new         |
+
+  Scenario Outline: delete a file when 2 files exist with different case
+    Given using <dav_version> DAV path
+    And user "user0" has uploaded file with content "uploaded content" to "/textfile1.txt"
+    And user "user0" has uploaded file with content "uploaded content" to "/TextFile1.txt"
+    Then the HTTP status code should be "204"
+    And as "user0" file "/textfile1.txt" should not exist
```
Comment (individual-it) [inline tests/acceptance/features/apiWebdavOperations/deleteFile.feature]: where does the delete happen?

**Review by individual-it — DISMISSED** [review]

File: tests/acceptance/features/apiWebdavOperations/deleteFile.feature (hunk 4 above)
Comment (individual-it) [reply tests/acceptance/features/apiWebdavOperations/deleteFile.feature]: I'm actually worried about the fact that CI passes

**Review by phil-davis — CHANGES_REQUESTED** [review]
This scenario fails for me locally. I don't know how it passed CI on drone - maybe related to the recent drone problems???

I suggest that each file should have different content. Then delete one of the files. Check that the deleted file does not exist, check that the remaining file has the expected content. That will confirm that both the correct file name and the correct file content were deleted.

**Review by phil-davis — APPROVED** [review]

**Comment by phil-davis** [comment]
@davitol when making the backport, please make sure that you have pulled the very latest ``stable10`` of core, and branch from that. Maybe the problems we have been having are related to PRs branched from old master or stable10 - I don't know.

**Comment by davitol** [comment]
Backport in : https://github.com/owncloud/core/pull/33633

**Pull request merged**